lib/install/validate/config_to_data.go

```diff
@@ -108,25 +108,45 @@ func setVCHResources(ctx context.Context, vch *object.ResourcePool, d *data.Data
 	}
 	cpu := p.Config.CpuAllocation.GetResourceAllocationInfo()
 	if cpu != nil {
+		HandleDefaultSettings(cpu)
 		setResources(&d.VCHCPULimitsMHz, &d.VCHCPUReservationsMHz, &d.VCHCPUShares, cpu)
 	}
 	memory := p.Config.MemoryAllocation.GetResourceAllocationInfo()
 	if memory != nil {
+		HandleDefaultSettings(memory)
 		setResources(&d.VCHMemoryLimitsMB, &d.VCHMemoryReservationsMB, &d.VCHMemoryShares, memory)
 	}
 	return nil
 }
 
-func setResources(limit *int, reservation *int, shares **types.SharesInfo, allocation *types.ResourceAllocationInfo) {
-	if limit != nil && allocation.Limit > -1 {
+func HandleDefaultSettings(allocation *types.ResourceAllocationInfo) {
+	if allocation == nil {
+		return
+	}
+	if allocation.Limit == -1 {
+		allocation.Limit = 0
+	}
+	if allocation.Reservation == 1 {
```

Comment:
During creation, to workaround govmomi omit empty value issue (which is generated from vsphere WSDL), we have to set reservation to 1 instead of 0.
But this 1 is not a custom setting, we don't want to show to that 1 in the user configuration, so reset that back to correct default.
Also add few comments in the code.